rust/src/format/page_table.rs

```diff
@@ -15,42 +15,56 @@
 // specific language governing permissions and limitations
 // under the License.
 
-use arrow_array::Int64Array;
+use arrow_array::builder::Int64Builder;
+use arrow_array::{Array, Int64Array};
 use arrow_schema::DataType;
-use std::collections::HashMap;
+use std::collections::BTreeMap;
+use tokio::io::AsyncWriteExt;
 
 use crate::encodings::plain::PlainDecoder;
 use crate::encodings::Decoder;
 use crate::error::Result;
 use crate::io::object_reader::ObjectReader;
+use crate::io::object_writer::ObjectWriter;
 
-#[derive(Debug)]
+#[derive(Clone, Debug, PartialEq)]
 pub struct PageInfo {
     pub position: usize,
     pub length: usize,
 }
 
+impl PageInfo {
+    pub fn new(position: usize, length: usize) -> Self {
+        Self { position, length }
+    }
+}
+
+/// Page lookup table.
 #[derive(Debug, Default)]
 pub struct PageTable {
-    pages: HashMap<i32, HashMap<i32, PageInfo>>,
+    pages: BTreeMap<i32, BTreeMap<i32, PageInfo>>,
```

Comment:
add a comment here to clarify this is  { field_id -> { batch_id -> page_info}} ?

Reply:
done